app/repositories/posts_repository.rb

```diff
@@ -0,0 +1,23 @@
+class PostsRepository
+  attr_reader :model
+
+  def initialize(model: PostContainer['post'])
```

Comment:
and remove this 
You can call "model" without initialize it 
Because you include container and call it "model"